src/dos/dos_classes.cpp

```
@@ -169,77 +169,84 @@ Bit32u DOS_InfoBlock::GetDeviceChain(void) {
-	sSave(sPSP,service[0],0xcd);
-	sSave(sPSP,service[1],0x21);
-	sSave(sPSP,service[2],0xcb);
+	SSET_BYTE(sPSP, exit[0], uint8_t(0xcd));
+	SSET_BYTE(sPSP, exit[1], uint8_t(0x20));
+	SSET_BYTE(sPSP, service[0], uint8_t(0xcd));
+	SSET_BYTE(sPSP, service[1], uint8_t(0x21));
+	SSET_BYTE(sPSP, service[2], uint8_t(0xcb));
 	/* psp and psp-parent */
-	sSave(sPSP,psp_parent,dos.psp());
-	sSave(sPSP,prev_psp,0xffffffff);
-	sSave(sPSP,dos_version,0x0005);
+	SSET_WORD(sPSP,psp_parent,dos.psp());
+	SSET_DWORD(sPSP, prev_psp, 0xffffffff);
+	SSET_WORD(sPSP, dos_version, uint16_t(0x0005));
 	/* terminate 22,break 23,crititcal error 24 address stored */
 	SaveVectors();
 
 	/* FCBs are filled with 0 */
 	// ....
 	/* Init file pointer and max_files */
-	sSave(sPSP,file_table,RealMake(seg,offsetof(sPSP,files)));
-	sSave(sPSP,max_files,20);
+	SSET_DWORD(sPSP, file_table, RealMake(seg, offsetof(sPSP, files)));
+	SSET_WORD(sPSP, max_files, uint16_t(20));
 	for (Bit16u ct=0;ct<20;ct++) SetFileHandle(ct,0xff);
 
 	/* User Stack pointer */
-//	if (prevpsp.GetSegment()!=0) sSave(sPSP,stack,prevpsp.GetStack());
+//	if (prevpsp.GetSegment()!=0) SSET_DWORD(sPSP,stack,prevpsp.GetStack());
 
 	if (rootpsp==0) rootpsp = seg;
 }
 
-Bit8u DOS_PSP::GetFileHandle(Bit16u index) {
-	if (index>=sGet(sPSP,max_files)) return 0xff;
-	PhysPt files=Real2Phys(sGet(sPSP,file_table));
-	return mem_readb(files+index);
+uint8_t DOS_PSP::GetFileHandle(uint16_t index) const
+{
+	if (index >= SGET_WORD(sPSP, max_files))
+		return 0xff;
+	PhysPt files = Real2Phys(SGET_DWORD(sPSP, file_table));
+	return mem_readb(files + index);
 }
 
-void DOS_PSP::SetFileHandle(Bit16u index, Bit8u handle) {
-	if (index<sGet(sPSP,max_files)) {
-		PhysPt files=Real2Phys(sGet(sPSP,file_table));
-		mem_writeb(files+index,handle);
+void DOS_PSP::SetFileHandle(uint16_t index, uint8_t handle)
+{
+	if (index < SGET_WORD(sPSP, max_files)) {
+		PhysPt files = Real2Phys(SGET_DWORD(sPSP, file_table));
+		mem_writeb(files + index, handle);
 	}
```

Thinking about the `else` scenario and how this function just eats the failure scenario and carries on.  
I would imagine that programs just crash further downstream from this when handles aren't set.
Maybe a `DEBUG_LOG_MSG` would help tighten up debugging if/when this happens?
(just an idea - not critical for this PR)

---

I agree - added `DEBUG_LOG_MSG` in here. It won't affect our release builds, so it's fine even if it'll end up never being triggered.